### ISSUE TITLE
File too long error in  build-namelist_test.pl

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#2408]
### Brief summary of bug

Running ./build-namelist_test.pl with the --compare option fails with a filename that's too long (for either perl or the filesystem?). It doesn't produce the file because of this problem so it can't compare it. Without the --compare option it just doesn't create the file.

### General bug information

**CTSM version you are using:** ctsm5.1.dev171 (probably exists before this)
**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Just the build-namelist tester

### Details of bug

`./build-namelist_test.pl --compare $SCRATCH/ctsm5.1.dev171/bld/unit_tester >& run2.out`

Results in the error below.

### Important output or errors that show the problem

From the output...

``` text
#   Failed test 'compare file user_nl_ctsm_real_parameters DNE for -phys+clm5_0 and -envxml_dir+.+-driver+mct+-lnd_frac++glade+campaign+cesm+cesmdata+cseg+inputdata+atm+datm7+domain.lnd.T31_gx3v7.090928.nc+-res+0.9x1.25+-ignore_ic_date+-clm_start_type+branch+-namelist+++a+nrevsn=+thing.nc++++-bgc+bgc+-crop'
#   at NMLTest/CompFiles.pm line 103.
#   Failed test 'compare file user_nl_ctsm_real_parameters DNE for -phys+clm5_0 and -envxml_dir+.+-driver+mct+-lnd_frac++glade+campaign+cesm+cesmdata+cseg+inputdata+atm+datm7+domain.lnd.T31_gx3v7.090928.nc+-res+0.9x1.25+-ignore_ic_date+-clm_start_type+startup+-namelist+++a+finidat=+thing.nc++++-bgc+bgc+-crop'
#   at NMLTest/CompFiles.pm line 103.
# Looks like you failed 2 tests of 3349.
.
.
.
177/3349 < FAIL> <Test Id: 177> <Desc: compare file user_nl_ctsm_real_parameters DNE for -phys+clm5_0 and -envxml_dir+.+-driver+mct+-lnd_frac++glade+campaign+cesm+cesmdata+cseg+inputdata+atm+datm7+domain.lnd.T31_gx3v7.090928.nc+-res+0.9x1.25+-ignore_ic_date+-clm_start_type+branch+-namelist+++a+nrevsn=+thing.nc++++-bgc+bgc+-crop>  
191/3349 < FAIL> <Test Id: 191> <Desc: compare file user_nl_ctsm_real_parameters DNE for -phys+clm5_0 and -envxml_dir+.+-driver+mct+-lnd_frac++glade+campaign+cesm+cesmdata+cseg+inputdata+atm+datm7+domain.lnd.T31_gx3v7.090928.nc+-res+0.9x1.25+-ignore_ic_date+-clm_start_type+startup+-namelist+++a+finidat=+thing.nc++++-bgc+bgc+-crop> 
```

**Comment by ekluzek** [comment]
Another simple thing I saw here is that the number of tests wasn't accurate, so I'm fixing that.

**Comment by ekluzek** [comment]
I don't think that people are using the file  user_nl_ctsm_real_parameters -- so one way to handle this would be to delete the option to create that file.

**Comment by ekluzek** [comment]
This has been handled in build-namelist_test.pl, but we should still remove the output_reals option in build-namelist.

**Comment by samsrabin** [comment]
Marking this as closed, with #2790 containing the only remaining task. Let me know if that's wrong, @ekluzek.
